nf_core/pipeline-template/main.nf

```diff
@@ -1,390 +1,63 @@
-    data: |
-        <dl class=\"dl-horizontal\">
-            $x
-        </dl>
-    """.stripIndent() }
-    .set { ch_workflow_summary }
+nextflow.enable.dsl = 2
 
 /*
- * Parse software version numbers
- */
-process get_software_versions {
-    publishDir "${params.outdir}/pipeline_info", mode: params.publish_dir_mode,
-        saveAs: { filename ->
-                      if (filename.indexOf('.csv') > 0) filename
-                      else null
-        }
-
-    output:
-    file 'software_versions_mqc.yaml' into ch_software_versions_yaml
-    file 'software_versions.csv'
+========================================================================================
+    GENOME PARAMETER VALUES
+========================================================================================
+*/
 
-    script:
-    // TODO nf-core: Get all tools to print their version number here
-    """
-    echo $workflow.manifest.version > v_pipeline.txt
-    echo $workflow.nextflow.version > v_nextflow.txt
-    fastqc --version > v_fastqc.txt
-    multiqc --version > v_multiqc.txt
-    scrape_software_versions.py &> software_versions_mqc.yaml
-    """
-}
+params.fasta = WorkflowMain.getGenomeAttribute(params, 'fasta')
 
 /*
- * STEP 1 - FastQC
- */
-process fastqc {
-    tag "$name"
-    label 'process_medium'
-    publishDir "${params.outdir}/fastqc", mode: params.publish_dir_mode,
-        saveAs: { filename ->
-                      filename.indexOf('.zip') > 0 ? "zips/$filename" : "$filename"
-        }
-
-    input:
-    set val(name), file(reads) from ch_read_files_fastqc
-
-    output:
-    file '*_fastqc.{zip,html}' into ch_fastqc_results
+========================================================================================
+    VALIDATE & PRINT PARAMETER SUMMARY
+========================================================================================
+*/
 
-    script:
-    """
-    fastqc --quiet --threads $task.cpus $reads
-    """
-}
+WorkflowMain.initialise(workflow, params, log)
 
 /*
- * STEP 2 - MultiQC
- */
-process multiqc {
-    publishDir "${params.outdir}/MultiQC", mode: params.publish_dir_mode
-
-    input:
-    file (multiqc_config) from ch_multiqc_config
-    file (mqc_custom_config) from ch_multiqc_custom_config.collect().ifEmpty([])
-    // TODO nf-core: Add in log files from your new processes for MultiQC to find!
-    file ('fastqc/*') from ch_fastqc_results.collect().ifEmpty([])
-    file ('software_versions/*') from ch_software_versions_yaml.collect()
-    file workflow_summary from ch_workflow_summary.collectFile(name: "workflow_summary_mqc.yaml")
+========================================================================================
+    NAMED WORKFLOW FOR PIPELINE
+========================================================================================
+*/
 
-    output:
-    file "*multiqc_report.html" into ch_multiqc_report
-    file "*_data"
-    file "multiqc_plots"
+workflow NFCORE_{{ short_name|upper }} {
```

Comment:
Why do we need to define that workflow here just to run it a bit further down? Can't we just run the workflow here already?
Seems unnecessary to me ...

Reply:
Ping @JoseEspinosa

Reply:
You can read part of the discussion on [this](https://github.com/nf-core/rnaseq/issues/619) issue of the rnaseq pipeline.
In short, this declaration allows calling a pipeline as a subworkflow. @ewels mentioned some of the uses that this could have on a Slack discussion:

*"eg. run RNAseq analysis and then run whatever custom different expression analysis. or run rnaseq + atacseq and do x + y afterwards, etc etc
Kind of like how the latest Snakemake release can run Nextflow workflows within a snakemake workflow
So this would be importing entire nf-core workflows into other, non-nf-core workflows that we would never see"*

Also, another scenario that could be used to include the [fetchngs](https://github.com/nf-core/fetchngs) pipeline in another pipeline as the first step to download the data. 

I am using it in a meta workflow that I am working on and that should allow to include a whole DSL2 pipeline to be benchmarked and that's the reason I proposed this change. 

Finally, it does not has any downside we came up in the previous discussion so that's why we agreed to change it.